src/context/wishlist-context.js

```diff
@@ -0,0 +1,107 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import { useCart } from "./cart-context";
+import { addItem } from "../services/addItem";
+import { removeItem } from "../services/removeItem";
+const WishlistContext = createContext();
+const initialState = {
+  items: [ ],
+  totalQuantity: 0,
+  error: null,
+  loading: false,
+};
+const WishlistProvider = ({ children }) => {
+  const [wishlist, setWishlist] = useState(initialState);
+  const setLoading = () =>
+    setWishlist((prev) => ({ ...prev, error: "", loading: true }));
+  const setError = (error) => {
+    setWishlist((prev) => ({ ...prev, error, loading: false }));
+  };
+  const updateWishlist = (updatedWishlist) => {
+    setWishlist((prev) => ({
+      ...prev,
+      items: [...updatedWishlist],
+      totalQuantity: [...updatedWishlist].length,
+      error: "",
+      loading: false,
+    }));
+  };
```

Comment:
It's better to use useReducer for all this instead of creating separate functions to update a specific part of state